File: PhotoBank.Dto/View/FilterDto.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PhotoBank.Dto.View
{
    public class FilterDto
    {
        public IEnumerable<int> Storages { get; set; }
        public bool? IsBW { get; set; }
        public bool? IsAdultContent { get; set; }
        public bool? IsRacyContent { get; set; }
        public string RelativePath { get; set; }
        public IEnumerable<int> Paths { get; set; }
        public string Caption { get; set; }
        public DateTime? TakenDateFrom { get; set; }
        public DateTime? TakenDateTo { get; set; }
        public bool? ThisDay { get; set; }
        public IEnumerable<int> Persons { get; set; }
        public IEnumerable<int> Tags { get; set; }
    }
}

File: Photobank.Ts/packages/shared/src/types/dto/FilterDto.ts
export interface FilterDto {
  storages?: number[];
  isBW?: boolean;
  isAdultContent?: boolean;
  isRacyContent?: boolean;
  relativePath?: string;
  paths?: number[];
  caption?: string;
  takenDateFrom?: string;
  takenDateTo?: string;
  thisDay?: boolean;
  persons?: number[];
  tags?: number[];
  orderBy?: string;
  skip?: number;
  top?: number;
}